media_driver/linux/common/ddi/media_libva.cpp

```diff
@@ -3289,6 +3289,22 @@ static VAStatus DdiMedia_SyncSurface (
     PDDI_DECODE_CONTEXT decCtx = (PDDI_DECODE_CONTEXT)surface->pDecCtx;
     if (decCtx && surface->curCtxType == DDI_MEDIA_CONTEXT_TYPE_DECODER)
     {
+        class MutexGuard {
+        private:
+            PMEDIA_MUTEX_T m_pMutex;
+        public:
+            MutexGuard(PMEDIA_MUTEX_T pMutex):m_pMutex(pMutex)
+            {
+                DdiMediaUtil_LockMutex(m_pMutex);
+            }
+            ~MutexGuard()
+            {
+                DdiMediaUtil_UnLockMutex(m_pMutex);
+            }
+        };
```

Comment:
Shouldn't that be somewhere in utilities?

Reply:
Fixed